Diagram.py

```diff
@@ -77,7 +77,27 @@ def listEntities(self) -> str:
         """
         entity_names = list(self._entities.keys())
         return ', '.join(entity_names)
-
-    def deleteRelation(self, relation: Relation) -> None:
-        pass
+        
+    def add_relation(self, source, destination):
```

Comment:
Please document the parameters for all of your methods. Since python doesn't require you to explicitly add the type, it is unclear as a reader what this parameters are.